### PR TITLE
Normalize Building MIEngine for VS/VSCode

### PR DESCRIPTION
**Pull request by WardenGnaw**
This PR adds ./eng/Scripts that can be used to build MIDebugEngine /
MIDebugEngine-Unix on Windows/macOS/Linux similar to how the CI does it.

CI-Test.[cmd|ps1] is not added because setup is a bit more complicated
than linux/macOS.

### CODE REVIEW

**Comment by gregg-miskelly** [comment]
In case this isn't obvious: do we want to update https://github.com/Microsoft/MIEngine/wiki/Building-the-MIEngine as well?

File: eng/Scripts/CI-Build.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+ScriptDir=`dirname $0`
```
Comment (gregg-miskelly) [inline eng/Scripts/CI-Build.sh:3]: >`dirname $0` [](http://example.com/codeflow?start=10&length=12)

shellcheck.net says you should use `ScriptDir=$(dirname "$0")`

**Comment by WardenGnaw** [comment]
> In case this isn't obvious: do we want to update https://github.com/Microsoft/MIEngine/wiki/Building-the-MIEngine as well?

Yep. I work working on MIEngine.wiki when I realized that we could simplify the process for folks.

File: eng/Scripts/CI-Build.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+RootDir=$ScriptDir/../..
+
+if [ "$(which dotnet)" == "" ]; then
+    echo "ERROR: Missing .NET SDK. Please install the SDK at https://dotnet.microsoft.com/download"
+fi
+
+dotnet build $RootDir/src/MIDebugEngine-Unix.sln
```
Comment (gregg-miskelly) [inline eng/Scripts/CI-Build.sh:10]: >$RootDir [](http://example.com/codeflow?start=13&length=8)

Shellcheck says you should quote rootdir

File: eng/Scripts/CI-Build.ps1 [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+$RootPath = Resolve-Path -Path "$PSScriptRoot\..\.."
+
+$msbuildPath = (Get-Command msbuild.exe -ErrorAction Ignore).Path
+
+if (!$msbuildPath) {
+    throw "Please run the script from a developer command prompt or have msbuild.exe in your PATH"
+}
+
+msbuild /t:Restore $RootPath\src\MIDebugEngine.sln /p:Configuration=$Configuration
```
Comment (gregg-miskelly) [inline eng/Scripts/CI-Build.ps1:24]: >msbuild [](http://example.com/codeflow?start=0&length=7)

Check exit code

File: eng/Scripts/CI-Build.ps1 [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+msbuild $RootPath\src\MIDebugEngine.sln /p:Configuration=$Configuration
```
Comment (gregg-miskelly) [inline eng/Scripts/CI-Build.ps1:25]: >msbuild [](http://example.com/codeflow?start=0&length=7)

Same

File: eng/Scripts/CI-Build.sh (hunk 2 above)
Comment (gregg-miskelly) [inline eng/Scripts/CI-Build.sh:10]: >dotnet [](http://example.com/codeflow?start=0&length=6)

Check exit code

File: eng/Scripts/CI-Build.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+$RootDir/PublishOpenDebugAD7.sh -c Debug -o $RootDir/bin/DebugAdapterProtocolTests/Debug/extension/debugAdapters
```
Comment (gregg-miskelly) [inline eng/Scripts/CI-Build.sh:11]: >$RootDir [](http://example.com/codeflow?start=0&length=8)

Same

File: eng/Scripts/CI-Test.sh [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+ScriptDir=$(dirname "$0")
+RootDir=$ScriptDir/../..
+
+if [ "$(which dotnet)" == "" ]; then
+    echo "ERROR: Missing .NET SDK. Please install the SDK at https://dotnet.microsoft.com/download"
+fi
+
+if [ ! -f "$RootDir/bin/DebugAdapterProtocolTests/Debug/CppTests/config.xml" ]; then
+    if [ "$(uname)" = "Darwin" ]; then
+        "$RootDir"/tools/DownloadLldbMI.sh "$RootDir"/bin/DebugAdapterProtocolTests/Debug/extension/debugAdapters
```
Comment (gregg-miskelly) [inline eng/Scripts/CI-Test.sh:12]: Check exit codes throughout

File: eng/Scripts/CI-Build.ps1 [hunk 7]
```diff
@@ -0,0 +1,39 @@
+param(
+[Alias("c")]
```
Comment (gregg-miskelly) [inline eng/Scripts/CI-Build.ps1:2]: >Alias [](http://example.com/codeflow?start=1&length=5)

Should we add a ValidationSet for this also?

**Review by gregg-miskelly — APPROVED** [review]

**Review by gregg-miskelly — APPROVED** [review]

**Pull request merged**